yellowbrick/classifier/class_balance_heat_map.py

```diff
@@ -0,0 +1,169 @@
+# Class balance visualizer for showing per-class support.
+#
+# Author:   Rebecca Bilbro <rbilbro@districtdatalabs.com>
+# Author:   Benjamin Bengfort <bbengfort@districtdatalabs.com>
+# Author:   Neal Humphrey
+# Author:   Larry Gray
+# Created:  Sun March 4 20:45:00 2018 -0400
+#
+# Copyright (C) 2017 District Data Labs
+# For license information, see LICENSE.txt
+#
+# ID: class_balance_heat_map.py [5388065] lwgray@gmail.com $
+
+"""
+Class balance heat map visualizer for showing per-class support.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import numpy as np
+
+from .base import ClassificationScoreVisualizer
+
+from sklearn.model_selection import train_test_split
+from sklearn.utils.multiclass import unique_labels
+
+
+##########################################################################
+## Class Balance Heat Map Chart
+##########################################################################
+
+class ClassBalanceHeatMap(ClassificationScoreVisualizer):
+    """
+    Class balance heat map that shows the support for each class in the
+    fitted classification model displayed as a stacked bar.  Each bar is
+    segmented to show the distribution of predicted classes for each
+    class. It is initialized with a fitted model and generates a
+    class balance heat map chart on draw.
+
+    Parameters
+    ----------
+    ax: axes
+        the axis to plot the figure on.
+    model: estimator
+        Scikit-Learn estimator object. Should be an instance of a classifier,
+        else ``__init__()`` will raise an exception.
+    classes: list
+        A list of class names for the legend. If classes is None and a y value
+        is passed to fit then the classes are selected from the target vector.
+    kwargs: dict
+        Keyword arguments passed to the super class. Here, used
+        to colorize the bars in the histogram.
+    Notes
+    -----
+    These parameters can be influenced later on in the visualization
+    process, but can and should be set as early as possible.
+    """
+
+    def score(self, X, y=None, **kwargs):
+        """
+        Generates a 2D array where each row is the count of the
+        predicted classes and each column is the true class
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        Returns
+        -------
+
+        ax : the axis with the plotted figure
+        """
+
+        y_pred = self.predict(X)
+        self.indices = unique_labels(y, y_pred)
+        self.classes_ = self.classes_ if self.classes_ is not None else self.indices
+        self.scores = np.array([
```

Comment:
Following the comment above, I'd say it is valid to store this as `self.scores_` -- many of the `*CV` estimators do something like this, though they compute the scores on fit. I think we can get away with that here. 

That also reminds me, there are several other visualizers that are not consistent with this behavior and need to be updated. @ndanielsen, @rebeccabilbro  - do you think we should make this an issue to deal with it in the future?

Reply:
I agree that we need to make an issue, and likely conduct an audit to ensure that all visualizers are conforming to this spec for self.score_ but also for things like self.classes_

Reply:
Ok - we'll have to make an issue for it; @lwgray would you mind doing that?

Reply:
Took care of it 👍 

#326